console-subscriber/src/builder.rs

```diff
@@ -149,6 +158,283 @@ impl Builder {
+///
+/// ## Further customization
+///
+/// To add additional layers or replace the format layer, replace
+/// `console_subscriber::init` with:
+///
+/// ```rust
+/// use tracing_subscriber::prelude::*;
+///
+/// let console_layer = console_subscriber::spawn();
+///
+/// tracing_subscriber::registry()
+///     .with(console_layer)
+///     .with(tracing_subscriber::fmt::layer())
+/// //  .with(..potential additional layer..)
+///     .init();
+/// ```
+///
+/// [`Targets`]: https://docs.rs/tracing-subscriber/latest/tracing-subscriber/filter/struct.Targets.html
+pub fn init() {
+    TasksLayer::builder().with_default_env().init();
+}
+
+/// Returns a new `tracing_subscriber` [`Layer`] configured with a [`TasksLayer`]
+/// and a [filter] that enables the spans and events required by the console.
+///
+/// This function spawns the console subscriber's [`Server`] in its own Tokio
+/// runtime in a background thread.
+///
+/// Unlike [`init`], this function does not set the default subscriber, allowing
+/// additional [`Layer`]s to be added.
+///
+/// [filter]: https://docs.rs/tracing-subscriber/latest/tracing_subscriber/filter/struct.Targets.html
+/// [`Layer`]: https://docs.rs/tracing-subscriber/latest/tracing_subscriber/layer/trait.Layer.html
+/// [`Server`]: crate::Server
+///
+/// ## Configuration
+///
+/// `console_subscriber::build` supports all of the environmental
+/// configuration described at [`console_subscriber::init`].
+///
+/// ## Differences from `init`
+///
+/// Unlike [`console_subscriber::init`], this function does *not* add a
+/// [`tracing_subscriber::fmt`] layer to the configured `Layer`. This means
+/// that this function will not log spans and events based on the value of the
+/// `RUST_LOG` environment variable. Instead, a user-provided [`fmt::Layer`] can
+/// be added in order to customize the log format.
+///
+/// You must call [`.init()`] on the final subscriber in order to [set the
+/// subscriber as the default][default].
+///
+/// ## Examples
+///
+/// ```rust
+/// use tracing_subscriber::prelude::*;
+/// tracing_subscriber::registry()
+///     .with(console_subscriber::spawn())
+///     .with(tracing_subscriber::fmt::layer())
+/// //  .with(...)
+///     .init();
+/// ```
+/// [`.init()`]: https://docs.rs/tracing-subscriber/latest/tracing_subscriber/util/trait.SubscriberInitExt.html
+/// [default]: https://docs.rs/tracing/latest/tracing/dispatcher/index.html#setting-the-default-subscriber
+/// [sub]: https://docs.rs/tracing/latest/tracing/trait.Subscriber.html
+/// [`tracing_subscriber::fmt`]: https://docs.rs/tracing-subscriber/latest/tracing-subscriber/fmt/index.html
+/// [`fmt::Layer`]: https://docs.rs/tracing-subscriber/latest/tracing-subscriber/fmt/struct.Layer.html
+/// [`console_subscriber::init`]: crate::init()
+#[must_use = "spawn() without init() will not set the default tracing subscriber"]
```

Comment:
should maybe change this as well:
```suggestion
#[must_use = "a `Layer` must be added to a `tracing::Subscriber`in order to be used"]
```